### ISSUE TITLE
Save iteration progress after crash?

### ISSUE THREAD

**Issue by guspolly** [austromorph/cartogram3#16]
QGIS will usually crash for me when I try to do a cartogram with lots of features and a large number of iterations (right now, for instance, I'm trying to do 100 iterations on 2850 features). I'm presuming it has something to do with memory or cpu or something. (I'm using cartogram 3.0 from the plugin manager on QGIS 3.0.3 on Windows 10.)

My main request though is, for instance, if it crashes during, say, iteration 30 of 100, can I get a dump of iteration 29? Because it might be good enough, but I don't want to run the whole thing again just to have it crash again on iteration 23 for instance.

**Comment by christophfink** [comment]
Yes that sounds like a memory issue. We currently do not test whether there is sufficient memory available – on Linux that did not every turn out to be a problem, though.
The output is a QGIS in-memory layer. It is – by definition – lost when QGIS crashes. I’ll think about how we could implement backup saving without affecting performance too heavily :)
